Add splunk-otel-collector secret to cluster receiver

tongpu opened the pull request:
Prior to this fix the cluster receiver wouldn't start because it was looking for the CA, but as it was not mounted it failed to start the process.

Would there be any other places where this might need to be added?

dmitryax commented on helm-charts/splunk-otel-collector/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: splunk-otel-collector
-version: 0.43.4
+version: 0.43.5
```
If you want to release it right away, please add a section to CHANGELOG and run `make render`

dmitryax replied:
Feel free to keep the current version. we will release it after

dmitryax reviewed (commented):
Thanks for fixing it!

dmitryax commented:
> Would there be any other places where this might need to be added?

I think it's also missed in gateway-deployment.yaml

tongpu commented:
> > Would there be any other places where this might need to be added?
> 
> I think it's also missed in gateway-deployment.yaml

I've added it to gateway-deployment.yaml as well and ran `make render` to render the static manifests. Should I updated the changelog as well?
